pkg/metrics/metrics.go

```diff
@@ -183,3 +185,90 @@ func (r *standardRegistry) ServiceRetriesCounter() metrics.Counter {
 func (r *standardRegistry) ServiceServerUpGauge() metrics.Gauge {
 	return r.serviceServerUpGauge
 }
+
+// ScalableHistogram is a Histogram with a predefined time unit used when
+// producing observations allowing the call to Start() and ObserveDuration()
+// without explicitly setting the observed value.
+type ScalableHistogram interface {
+	With(labelValues ...string) ScalableHistogram
+	Start()
+	Observe(v float64)
+	ObserveDuration()
+}
```

Comment:
Why is it useful to have this interface? as far as I can see, there's only one implementation for it (aside from histogramMock), which is HistogramWithScale, right? So why don't we simply have HistogramWithScale everywhere?

Reply:
Initially I thought about tests and trying to keep it interchangeable with Timer, but it didn't work out :)
So yes we can have HistogramWithScale directly if you prefer

Reply:
yes, I don't think the interface brings anything to the table, and it looks like it makes the whole change a little bit more complicated than it needs to be.

Reply:
actually the interface is used to hide `MultiHistogram` (slice of Histograms) as well. In true all metric objects on `standardRegistry` are slices. I think it will be better to keep the same pattern in this case. Do we keep it or proceed to expose a concrete type only for histograms?

Reply:
ah ok. If both the type and the slice of type are indeed publicly needed, then yes, I suppose it's fair enough to keep them behind the interface, if that makes the code simpler.